### PR TITLE
Fix equity account creation for different commodities

### PR DESCRIPTION
**Pull request by rhabacker**
CCBUG:798163

### CODE REVIEW

File: gnucash/gnome-utils/dialog-account.c [hunk 1]
```diff
@@ -526,7 +526,13 @@ gnc_ui_to_account(AccountWindow *aw)
 
     if (use_equity)
     {
-        if (!gnc_account_create_opening_balance (account, balance, date, aw->book))
+        if (xaccAccountGetType (account) == ACCT_TYPE_STOCK ||
+            xaccAccountGetType (account) == ACCT_TYPE_MUTUAL)
+            commodity = xaccAccountGetCommodity (parent_account);
+        else
+            commodity = xaccAccountGetCommodity (account);
```
Comment (christopherlam) [inline gnucash/gnome-utils/dialog-account.c:533]:
```suggestion
        commodity = gnc_account_get_currency_or_parent (account);
```

File: gnucash/gnome/assistant-hierarchy.c [hunk 2]
```diff
@@ -1404,13 +1404,31 @@ static void
...
+    {
+        Account *parent_account = gnc_account_get_parent (account);
+        if (parent_account == NULL)
+            parent_account = gnc_get_current_root_account ();
+        commodity = xaccAccountGetCommodity (parent_account);
+    }
+    else
+    {
+        commodity = xaccAccountGetCommodity (account);
+    }
```
Comment (christopherlam) [inline gnucash/gnome/assistant-hierarchy.c:1428]:
```suggestion
    commodity = gnc_account_get_currency_or_parent (account);
```

**Comment by jralls** [comment]
@rhabacker This doesn't seem to change any behavior.

**Comment by rhabacker** [comment]
The problem has already been fixed with https://github.com/Gnucash/gnucash/commit/267a7cb75d3b54cf359eb913c9f830b48307ae6a.

**Pull request closed without merging**